google/cloud/bigtable/data/mutations.py

```diff
@@ -36,6 +39,12 @@ class Mutation(ABC):
     def _to_dict(self) -> dict[str, Any]:
         raise NotImplementedError
 
+    def _to_pb(self) -> data_pb.Mutation:
```

Comment:
Do we need to double store all of the attributes? Can't we store all of the attributes in the proto directly?
Also can we drop _to_dict altogether?

Reply:
I think the main downside is we'd need a lot more boilerplate setters/getters (we lose the simplicity of the dataclasses). And marshaling to/from the protos is more expensive, but I guess that should'nt be too much of an issue here. I see this is how we handled it in the [ReadRowsQuery class](https://github.com/googleapis/python-bigtable/blob/experimental_v3/google/cloud/bigtable/data/read_rows_query.py#L184), which is more complicated than these would be.

What do you think of making these immutable? That would simplify the setters/getters, and then we wouldn't have to worry about making a static copy before starting a mutate_rows operation

Reply:
> Also can we drop _to_dict altogether?

I think so. We do use `row._to_dict` in the test proxy, but I don't think we'd need it for mutations

Reply:
ideally the logic for converting from models to protos for the test proxy is self contained in the proxy. A side goal of test proxy was to be a rosetta stone for expressing the same concepts across every cleint impl